packages/rum-vue/src/route-hooks.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import { Promise } from 'es6-promise'
+
+export function routeHooks(router, apm) {
+  let transaction
+
+  function scheduleTransactionFinish() {
+    if (transaction) {
+      Promise.resolve().then(() => transaction.detectFinish())
+    }
+  }
+
+  router.beforeEach((to, from, next) => {
```

Comment:
I think we might need both `beforeEach` and `beforeResolve`.  The reason is that during `beforeEach` callbacks they might decide to move to another route instead of the one that initiated the original event. For example if the user is not logged in instead of going to the actual route it would go to the login route.

We probably need test for this scenario as well.

Reply:
Decided to keep the current approach and test if it's actually a problem, separately